docs/ibc/apps/ibcmodule.md

```diff
@@ -0,0 +1,350 @@
+
+```go
+// Called by IBC Handler on MsgCloseInit
+func (im IBCModule) OnChanCloseInit(
+    ctx sdk.Context,
+    portID,
+    channelID string,
+) error {
+    // ... do custom finalization logic
+
+    // Use above arguments to determine if we want to abort handshake
+    err := checkArguments(args)
+    return err
+}
+
+// Called by IBC Handler on MsgCloseConfirm
+func (im IBCModule) OnChanCloseConfirm(
+    ctx sdk.Context,
+    portID,
+    channelID string,
+) error {
+    // ... do custom finalization logic
+
+    // Use above arguments to determine if we want to abort handshake
+    err := checkArguments(args)
+    return err
+}
+```
+
+### Channel handshake version negotiation
+
+Application modules are expected to verify versioning used during the channel handshake procedure.
+
+- `onChanOpenInit` will verify that the relayer-chosen parameters
+  are valid and perform any custom `INIT` logic.
+  It may return an error if the chosen parameters are invalid
+  in which case the handshake is aborted.
+  If the provided version string is non-empty, `onChanOpenInit` should return
+  the version string if valid or an error if the provided version is invalid.
+  **If the version string is empty, `onChanOpenInit` is expected to
+  return a default version string representing the version(s)
+  it supports.**
+  If there is no default version string for the application,
+  it should return an error if provided version is empty string.
+- `onChanOpenTry` will verify the relayer-chosen parameters along with the
+  counterparty-chosen version string and perform custom `TRY` logic.
+  If the relayer-chosen parameters
+  are invalid, the callback must return an error to abort the handshake.
+  If the counterparty-chosen version is not compatible with this modules
+  supported versions, the callback must return an error to abort the handshake.
+  If the versions are compatible, the try callback must select the final version
+  string and return it to core IBC.
+  `onChanOpenTry` may also perform custom initialization logic
+- `onChanOpenAck` will error if the counterparty selected version string
+  is invalid to abort the handshake. It may also perform custom ACK logic.
+
+Versions must be strings but can implement any versioning structure. If your application plans to
+have linear releases then semantic versioning is recommended. If your application plans to release
+various features in between major releases then it is advised to use the same versioning scheme
+as IBC. This versioning scheme specifies a version identifier and compatible feature set with
+that identifier. Valid version selection includes selecting a compatible version identifier with
+a subset of features supported by your application for that version. The struct is used for this
+scheme can be found in `03-connection/types`.
```

Comment:
Do you mean [this](https://github.com/cosmos/ibc-go/blob/main/modules/core/03-connection/types/version.go#L16)? Should we maybe add the link here as well?